src/xc_integrator/replicated/host/reference_replicated_xc_host_integrator_exc_vxc.hpp

```diff
@@ -255,14 +292,56 @@ void ReferenceReplicatedXCHostIntegrator<ValueType>::
       dden_z_eval   = dden_y_eval + spin_dim_scal * npts;
     }
 
+    if ( func.is_mgga() ) {
+      dbasis_x_eval = basis_eval    + npts * nbe;
+      dbasis_y_eval = dbasis_x_eval + npts * nbe;
+      dbasis_z_eval = dbasis_y_eval + npts * nbe;
+      dden_x_eval   = den_eval    + spin_dim_scal * npts;
+      dden_y_eval   = dden_x_eval + spin_dim_scal * npts;
+      dden_z_eval   = dden_y_eval + spin_dim_scal * npts;
+      mmat_x        = mmat;
+      mmat_y        = mmat_x + npts * nbe;
+      mmat_z        = mmat_y + npts * nbe;
+      // TODO: Add check for Laplacian-dependent functionals
+      if ( true ) {
+	d2basis_xx_eval = dbasis_z_eval + npts * nbe;
+	d2basis_xy_eval = d2basis_xx_eval + npts * nbe;
+	d2basis_xz_eval = d2basis_xy_eval + npts * nbe;
+	d2basis_yy_eval = d2basis_xz_eval + npts * nbe;
+	d2basis_yz_eval = d2basis_yy_eval + npts * nbe;
+	d2basis_zz_eval = d2basis_yz_eval + npts * nbe;
+	lbasis_eval     = d2basis_zz_eval + npts * nbe;
+      }
+      if(is_uks) {
+	mmat_x_z = mmat_z + npts * nbe;
+	mmat_y_z = mmat_x_z + npts * nbe;
+	mmat_z_z = mmat_y_z + npts * nbe;
+      }
+    }
+
 
     // Get the submatrix map for batch
     std::vector< std::array<int32_t, 3> > submat_map;
     std::tie(submat_map, std::ignore) =
           gen_compressed_submat_map(basis_map, task.bfn_screening.shell_list, nbf, nbf);
 
+    // Evaluate Collocation (+ Grad and Hessian)
+    if( func.is_mgga() ) {
+      if ( true ) {
+        lwd->eval_collocation_hessian( npts, nshells, nbe, points, basis, shell_list,
```

Comment:
We might be able to hack `gau2grid` to quickly evaluate the laplacian as opposed to the full hessian. Would definitely cut down on memory and should make the evaluation a bit faster (50% max)

Reply:
Agreed

Reply:
Unless you want to add this, we can punt for a bit. I've added an issue (https://github.com/wavefunction91/GauXC/issues/85) to track

Reply:
Let's punt it